### PR TITLE
Ops in own classes so persistent vars possible.

### PR DESCRIPTION
**Pull request by paulprozesky**
@brickZA - can we do this?

### CODE REVIEW

File: src/fxcorrelator.py [hunk 1]
```diff
@@ -62,6 +64,7 @@ def __init__(self, descriptor, identifier=-1, config_source=None,
 
         """
         self.logger = logger
+        self.loghandler = None
```
Comment (brickZA) [inline src/fxcorrelator.py]: Is this needed?

Comment (paulprozesky) [inline src/fxcorrelator.py]: PyCharm says I should not assign class attributes outside of init. What do you think?

**Comment by paulprozesky** [comment]
So another way to do this would be to keep all the ops functions as they were, but have config dictionaries that belong to the corr instance for x/f/b-specific runtime attributes?

So f-eng ops could write to and read from corr.feng_vars['n_chans'] for example?

File: src/fxcorrelator.py (hunk 1 above)
Comment (brickZA) [inline src/fxcorrelator.py]: Oh, so that is set in the default log config method? I guess it's fine then.

**Comment by brickZA** [comment]
@paulprozesky I think it is nice to clean separation between layers, e.g. it might be good if the fengops does not have to understand the format of the config file, so I like the idea of have the fxcorrelator object parsing the config file and then making it available in a standard format.

In any case, the PR as it stands now looks fine. I have not checked each line in detail for potential issues, but the general approach looks fine. As far as I'm concerned, merge at will :)

**Pull request merged**